### PR TITLE
Add onchange to deflisten

### PR DESCRIPTION
**Pull request by owenrumney**
## Description

Adding onchange to deflisten as per #355 so that I can pop out the spotify expander.

I think this has other uses for things like a deflisten that watches for github stars or PRs and triggers something on the machine.

## Usage

Add an `onchange` attribute to the `deflisten` with the command you want to run.

```
(deflisten music
  :initial ""
  :onchange "eww update revealSpotify=true && sleep 5 && eww update revealSpotify=false"
"playerctl --follow metadata --format '{{ artist }} - {{ title }}' || true")
```

### Showcase

listenonchange

## Additional Notes

This is Day0 of writing Rust, if it is totally off the mark or not Rustic then I can change as required.

I ran `cargo fmt` and it changed loads of files :eyes:

## Checklist

Please make sure you can check all the boxes that apply to this PR.

- [x] All widgets I've added are correctly documented.
- [x] I added my changes to CHANGELOG.md, if appropriate.
- [x] The documentation in the `docs/content/main` directory has been adjusted to reflect my changes.
- [x] I used `cargo fmt` to automatically format all code before committing

### CODE REVIEW

**Comment by elkowar** [comment]
I don't quite see the utility here, when you could also simply do
```
playerctl --follow metadata --format '{{ artist }} - {{ title }}' | while read -r line; do
  echo $line
  { eww update revealSpotify=true; sleep 5; eww update revealSpotify=false } &
done
```
without requiring any specific support form eww

**Comment by owenrumney** [comment]
Yeah, that's valid I guess! I also use it with a small utility listening for dbus notifications and it slides out an expander with the message. From a usage perspective, I think it's more clear that when the variable changes something explicit will happen, rather than all wrapped in the listening command.

TL;DR; I take your point, happy for you to reject if you don't see a place for it in upstream.

**Comment by elkowar** [comment]
Yea, I'm afraid I don't think this is really necessary, and would probably add more confusion than anything else. In the future, there will be some changes that allow you to change ewws state more directly from within eww, in which case this would maybe be a useful addition. Until that is in place however I don't think this provides enough utiIty. Glad you understand, and still very much thanks for your contribution!
